runtime/src/main/java/org/corfudb/runtime/collections/LRMultiNamespaceListener.java

```diff
@@ -0,0 +1,86 @@
+package org.corfudb.runtime.collections;
+
+import lombok.Setter;
+import org.corfudb.runtime.LogReplication.ReplicationStatusVal;
+import org.corfudb.runtime.view.ObjectsView;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+
+/**
+ * This is the callback interface that any client subscribing to updates across the LR Status Table
+ * (ReplicationStatus) and application-specific tables must implement.
+ *
+ * This interface sees ordered updates from :
+ * 1. client-streams from client-Namespace, and,
+ * 2. LrStatusTable from corfuSystem-Namespace.
+ */
+public abstract class LRMultiNamespaceListener implements StreamListener {
+
+    private boolean snapshotSyncInProgress = false;
+
+    private static final String REPLICATION_STATUS_TABLE = ObjectsView.REPLICATION_STATUS_TABLE;
+
+    @Setter
+    long subscriptionTs = 0;
+
+    public void onNextEntry(CorfuStreamEntries results) {
```

Comment:
This has the same behavior as default, no need to rewrite, right?

Reply:
yes, that's right